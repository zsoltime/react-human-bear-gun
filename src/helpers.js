export const stringOrNull = (props, propName, componentName) => {
  const propValue = props[propName];

  if (propValue === null || typeof propValue === 'string') {
    return;
  }

  return new Error(
    `Invalid prop \`${propName}\` of type \`${typeof propValue}\` supplied to \`${componentName}\`, expected \`string\` or \`null\`.`
  );
};
