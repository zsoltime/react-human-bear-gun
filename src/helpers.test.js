import { stringOrNull } from './helpers';

describe('stringOrNull prop type', () => {
  it('returns undefined if prop is string or null', () => {
    const props = {
      str: 'Test string',
      nil: null,
    };

    expect(stringOrNull(props, 'str', 'TestComponent')).toBe(
      undefined
    );
    expect(stringOrNull(props, 'nil', 'TestComponent')).toBe(
      undefined
    );
  });

  it('returns new Error if prop is not a string or not null', () => {
    const props = {
      truthyBool: true,
      falsyBool: false,
      fn: () => null,
      num: 57,
      obj: { a: 5, b: null },
      sym: Symbol('sym'),
    };

    for (const propName of Object.keys(props)) {
      expect(
        stringOrNull(props, propName, 'TestComponent')
      ).toBeInstanceOf(Error);
      expect(
        stringOrNull(props, propName, 'TestComponent').message
      ).toMatchSnapshot();
    }
  });
});
