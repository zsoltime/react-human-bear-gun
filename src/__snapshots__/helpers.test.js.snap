// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringOrNull prop type returns new Error if prop is not a string or not null 1`] = `"Invalid prop \`truthyBool\` of type \`boolean\` supplied to \`TestComponent\`, expected \`string\` or \`null\`."`;

exports[`stringOrNull prop type returns new Error if prop is not a string or not null 2`] = `"Invalid prop \`falsyBool\` of type \`boolean\` supplied to \`TestComponent\`, expected \`string\` or \`null\`."`;

exports[`stringOrNull prop type returns new Error if prop is not a string or not null 3`] = `"Invalid prop \`fn\` of type \`function\` supplied to \`TestComponent\`, expected \`string\` or \`null\`."`;

exports[`stringOrNull prop type returns new Error if prop is not a string or not null 4`] = `"Invalid prop \`num\` of type \`number\` supplied to \`TestComponent\`, expected \`string\` or \`null\`."`;

exports[`stringOrNull prop type returns new Error if prop is not a string or not null 5`] = `"Invalid prop \`obj\` of type \`object\` supplied to \`TestComponent\`, expected \`string\` or \`null\`."`;

exports[`stringOrNull prop type returns new Error if prop is not a string or not null 6`] = `"Invalid prop \`sym\` of type \`symbol\` supplied to \`TestComponent\`, expected \`string\` or \`null\`."`;
